raven/raven.go

```diff
@@ -268,3 +306,29 @@ func (T *transport) RoundTrip(req *http.Request) (*http.Response, error) {
 	defer timer.Stop()
 	return T.httpTransport.RoundTrip(req)
 }
+
+type EventEncoder interface {
```

Comment:
Yup, I agree with you and I don't really have any strong feelings toward any of the approaches.
I was think that maybe we could make encode a method on Event, this would make it possible to use the same approach for testing I think, and wouldn't add a lot of inderection. Thoughts about that?